dataland-frontend/src/components/forms/CreateEuTaxonomyNonFinancials.vue

```diff
@@ -327,9 +330,24 @@ export default defineComponent({
         }
 
         const euTaxonomyForNonFinancialsDataControllerApi = this.buildEuTaxonomyNonFinancialsDataApi();
-        await euTaxonomyForNonFinancialsDataControllerApi!.postFrameworkData(
-          this.companyAssociatedEutaxonomyNonFinancialsData
+
+        const isCompanyOwner = await hasUserCompanyRoleForCompany(
+          CompanyRole.CompanyOwner,
+          this.companyAssociatedEutaxonomyNonFinancialsData.companyId,
+          this.getKeycloakPromise
         );
+
+        if (isCompanyOwner) {
+          await euTaxonomyForNonFinancialsDataControllerApi!.postFrameworkData(
+            this.companyAssociatedEutaxonomyNonFinancialsData,
```

Comment:
Könnte man in den ganzen Create Komponenten die Logik ob man CompanyOwner ist nicht auslagern in eine eigene Funktion, die einen Boolean zurückgibt. Diesen Boolean könnte man direkt in den .postFramworkData Call packen. Dann spart man sich die IF-Abfrage und Duplicate Code.

Reply:
the code has been refactored

Reply:
ok